### PR TITLE
Update Logger

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: include/veriblock/logger.hpp [hunk 1]
```diff
@@ -88,12 +88,12 @@ LogLevel StringToLevel(const std::string&);
...
 //! @private
 #define VBK_LOG(lvl, format, ...)                             \
-  do {                                                        \
+  {                                                           \
     auto& logger = altintegration::GetLogger();               \
     if (logger.level <= lvl) {                                \
       logger.log(lvl, VBK_LOG_FORMAT(format, ##__VA_ARGS__)); \
     }                                                         \
-  } while (0)
+  }
```
Comment (Warchant) [inline include/veriblock/logger.hpp:96]: do not remove these. it is the only way to make VBK_LOG work as expression in C++

File: include/veriblock/logger.hpp [hunk 2]
```diff
@@ -59,12 +59,12 @@ struct Logger {
...
 Logger& GetLogger();
 //! setter for global logger instance
-void SetLogger(std::unique_ptr<Logger> lgr);
+void SetLogger(std::unique_ptr<Logger> lgr, LogLevel log_lvl);
 
 //! setter for global logger instance
 template <typename L>
-void SetLogger() {
-  SetLogger(std::unique_ptr<L>(new L()));
+void SetLogger(LogLevel log_lvl) {
```
Comment (Warchant) [inline include/veriblock/logger.hpp:66]: don't add log level here please.
Instead, 
```
altintegration::GetLogger().level = yourlevel;
```

File: include/veriblock/logger.hpp [hunk 3]
```diff
@@ -93,7 +93,7 @@ LogLevel StringToLevel(const std::string&);
     if (logger.level <= lvl) {                                \
       logger.log(lvl, VBK_LOG_FORMAT(format, ##__VA_ARGS__)); \
     }                                                         \
-  } while (0)
+  } while (false);
```
Comment (Warchant) [inline include/veriblock/logger.hpp:96]:
```suggestion
  } while (false)
```

Comment (Mr-Leshiy) [reply include/veriblock/logger.hpp]: fixed

**Pull request merged**